numpy/lib/function_base.py

```diff
@@ -2669,6 +2689,18 @@ class vectorize(object):
            <http://docs.scipy.org/doc/numpy/reference/c-api.generalized-ufuncs.html>`_.
     """
 
+    def __new__(cls, pyfunc=None, *args, **kw):
```

Comment:
If I were doing this fresh, i'd still probably do it this way, because exposing the type is convenient.